libgalois/include/katana/GraphTopology.h

```diff
@@ -709,7 +709,8 @@ class KATANA_EXPORT ProjectedTopology : public GraphTopologyTypes {
 template <typename Topo>
 class KATANA_EXPORT BasicTopologyWrapper : public GraphTopologyTypes {
 public:
-  explicit BasicTopologyWrapper(const Topo* t) noexcept : topo_ptr_(t) {
+  explicit BasicTopologyWrapper(const std::shared_ptr<const Topo>& t) noexcept
+      : topo_ptr_(t) {
```

Comment:
think convention is to pass by value and move, which allows users to move their shared pointer into the constructor if they are done with it.

Reply:
Good point.